File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0;
        int l = 0;
        int r = height.size() - 1;

        while(l <= r){
            maxArea = std::max(maxArea, (r - l) * std::min(height[l], height[r]));
            height[l] > height[r] ? r-- : l++;
        }

        return maxArea;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {
    lo := 0
    hi := len(height) - 1

    max := 0

    for lo < hi{

        l := hi - lo
        h := 0

        if height[lo] < height[hi]{
            h = height[lo]
            lo++
        }else{
            h = height[hi]
            hi--
        }

        v := l * h

        if v > max{
            max = v
        }
    }

    return max
}